include/ppx/grfx/grfx_pipeline.h

```diff
@@ -243,6 +227,24 @@ struct PipelineInterfaceCreateInfo
         uint32_t                         set     = PPX_VALUE_IGNORED; // Set number
         const grfx::DescriptorSetLayout* pLayout = nullptr;           // Set layout
     } sets[PPX_MAX_BOUND_DESCRIPTOR_SETS] = {};
+
+    // VK: Push constants
+    // DX: Root constants
+    //
+    // Push/root constants are measured in DWORDs (uint32_t) aka 32-bit values.
+    //
+    // The binding and set for push constants CAN NOT overlap with a binding
+    // AND set in sets (the struct immediately above this one). It's okay for
+    // push constants to be in an existing set at binding that is not used
+    // by an entry in the set layout.
+    //
+    struct
+    {
+        uint32_t              count           = 0;                 // Measured in DWORDs, must be less than or equal to PPX_MAX_PUSH_CONSTANTS
+        uint32_t              binding         = PPX_VALUE_IGNORED; // D3D12 only, ignored by Vulkan
```

Comment:
Not very familiar with DXC register/binding relationship.
If I add a second ConstantBuffer in the shader, this time bound to the register `b9`,
DXC seems to say I have a constant buffers, bound to `cb0` and `cb9`.
So I'd have 2 bindings to do here, 1 with a binding to 0, 1 with a binding set to 9 no?

On the Vulkan side, seems like we can only have 1 push constant, so should BigWheels block this to 1 too on the DXC side? and maybe assert that binding is 0?
It might be restrictive, but at least is we compile a sample on windows+dx, we'd assert, and could learn "oh, vulkan doesn't support that, to be portable I need to do X?"

(Same reason we block the constant count to 32?)

Reply:
> Not very familiar with DXC register/binding relationship. If I add a second ConstantBuffer in the shader, this time bound to the register `b9`, DXC seems to say I have a constant buffers, bound to `cb0` and `cb9`. So I'd have 2 bindings to do here, 1 with a binding to 0, 1 with a binding set to 9 no?

Do you mean DX here or is DXC on purpose? The example you gave is correct, there would be two bindings at 1 and 9. I think I'm misunderstanding something here. Can you add some more detail?

> 
> On the Vulkan side, seems like we can only have 1 push constant, so should BigWheels block this to 1 too on the DXC side? and maybe assert that binding is 0? It might be restrictive, but at least is we compile a sample on windows+dx, we'd assert, and could learn "oh, vulkan doesn't support that, to be portable I need to do X?"

I'm not able to follow, can you give an example?

Reply:
DX for DirectX, sorry, too used to say DXC..
After looking at the PipelineInterfaceCreateInfo, we do enforce 1 pushConstant per pipeline anyway, so my concern is not valid.
One could create 2 pushconstant in a shader, which is valid on DirectX, but won't be able to use them both in BigWheels, so that's fine.

Was worried we could do something in DirectX which was invalid in Vulkan. But no. Thanks!